### PR TITLE
Task/14824 new thank you tile on home screen

### PR DESCRIPTION
**Pull request by 30mar**
## Description
Added new thank you tile as per tech-specs

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14824

## Screenshots
https://user-images.githubusercontent.com/15270737/220324371-3076a7b1-49f3-47a1-a04c-bac35ecabb77.mov

### CODE REVIEW

**Comment by Ein-Tim** [comment]
The file upload for the videos wasn't successful, unfortunately. Maybe you could try again? (:

**Comment by 30mar** [comment]
@Ein-Tim thanks for the heads up, i re uploaded the video

File: src/xcode/ENA/ENA.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -968,6 +968,9 @@
 		8F3D71AA25A86AD300D52CCD /* HomeExposureLoggingCellModelTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8F3D71A925A86AD300D52CCD /* HomeExposureLoggingCellModelTests.swift */; };
 		8F43A3CB25D55A3E008FD630 /* ppa_data.pb.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8F43A3C925D55A3D008FD630 /* ppa_data.pb.swift */; };
 		8F43A3CC25D55A3E008FD630 /* ppa_data_request_ios.pb.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8F43A3CA25D55A3E008FD630 /* ppa_data_request_ios.pb.swift */; };
+		8F43E69029A32A890047C21F /* EndOfLifeThankYouCell.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8F43E68E29A32A890047C21F /* EndOfLifeThankYouCell.swift */; };
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA.xcodeproj/project.pbxproj:971]: Funny name 😂

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/EndOfLifeThankYou/EndOfLifeThankYouCell.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+		setupAccessibility()
+	}
+	
+	private func updateIllustration(for traitCollection: UITraitCollection) {
+		if traitCollection.preferredContentSizeCategory >= .accessibilityLarge {
+			illustrationImageView.isHidden = true
+		} else {
+			illustrationImageView.isHidden = false
+		}
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/EndOfLifeThankYou/EndOfLifeThankYouCell.swift]:
```suggestion
illustrationImageView.isHidden = traitCollection.preferredContentSizeCategory >= .accessibilityLarge
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/EndOfLifeThankYou/EndOfLifeThankYouCellViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+//
+
+import UIKit
+
+class EndOfLifeThankYouCellViewModel {
+
+	// MARK: - Internal
+
+	var title = AppStrings.Home.EndOfLifeThankYouTile.title
+	// TODO get the xx value in figma
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/EndOfLifeThankYou/EndOfLifeThankYouCellViewModel.swift]:
```suggestion
	// to.do get the xx value in figma
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/HomeTableViewController.swift [hunk 4]
```diff
@@ -106,8 +106,16 @@ class HomeTableViewController: UITableViewController, NavigationBarOpacityDelega
 		viewModel.cclService.shouldShowNoticeTile
 			.receive(on: DispatchQueue.OCombine(.main))
 			.sink { [weak self] shouldShowNoticeTile in
+				let isHibernationState = CWAHibernationProvider.shared.isHibernationState
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Scenes/Home/HomeTableViewController.swift:109]: Please pass the dependency in the constructor

```swift
init(cwaHibernationProvider: CWAHibernationProvider = CWAHibernationProvider.shared) { ... }
```

**Review by flxschmidt — CHANGES_REQUESTED** [review]
Nice code, clean. Minor hints. :)

File: src/xcode/ENA/ENA/Source/Scenes/Home/HomeTableViewController.swift (hunk 4 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Scenes/Home/HomeTableViewController.swift:109]: But it is a global variable why do we need to inject it?

File: src/xcode/ENA/ENA.xcodeproj/project.pbxproj (hunk 1 above)
Comment (30mar) [reply src/xcode/ENA/ENA.xcodeproj/project.pbxproj:971]: we had another thankyou screen a year ago, and we also have  another endoflife tile "for the notice" so im trying to be more specific 😃  but do you have a better suggestion ?

File: src/xcode/ENA/ENA/Source/Scenes/Home/HomeTableViewController.swift (hunk 4 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/HomeTableViewController.swift:109]: Cause of a clean pattern style. In this way, you could also write a unit test and pass a mocked version into it. Or a version with a mocked store into it. Thatswhy I wrote a second constructor for the singleton, that can take a mocked store. ;)

Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/HomeTableViewController.swift:109]: As discussed, you can keep the singleton instance getter, without dependency injection as long you don't write a specific unit test with a mocked storage instance.

**Review by KathrinLuecke — APPROVED** [review]
UA approved

**Review by flxschmidt — APPROVED** [review]
LGTM

**Review by naveeddotio — APPROVED** [review]

**Review by puneetmahali1 — APPROVED** [review]

**Pull request merged**